jetson/nav/search/searchStateMachine.cpp

```diff
@@ -213,19 +209,41 @@ NavState SearchStateMachine::executeTurnToTarget()
 {
     if( mRover->roverStatus().leftTarget().distance < 0 )
     {
-        cerr << "Lost the target. Continuing to turn to last known angle\n";
-        if( mRover->turn( mTargetAngle + mTurnToTargetRoverAngle ) )
-        {
-            return NavState::TurnedToTargetWait;
+        // TODO: Clean this up and/or move it to avoid duplication
+        if ( mRover->roverStatus().leftCacheTarget().distance != -1 ) {
+            if( mRover->turn( mRover->roverStatus().leftCacheTarget().bearing +
+                      mRover->roverStatus().odometry().bearing_deg ) )
+            {
+                return NavState::DriveToTarget;
+            }
+            updateTargetDetectionElements( mRover->roverStatus().leftCacheTarget().bearing,
+                                       mRover->roverStatus().odometry().bearing_deg );
+            return NavState::TurnToTarget;
+        }
+        else {
+            cerr << "Lost the target. Continuing to turn to last known angle\n";
+            if( mRover->turn( mTargetAngle + mTurnToTargetRoverAngle ) )
+            {
+                return NavState::TurnedToTargetWait;
+            }
+            return NavState::TurnToTarget;
         }
-        return NavState::TurnToTarget;
     }
-    if( mRover->turn( mRover->roverStatus().leftTarget().bearing +
+
+    // Consider if we have a valid cache. If we do, use it. Otherwise, 
+    // can just use the regular leftTarget since it will be filled or empty
```

Comment:
We need to flush it out at somepoint, so I felt after sometime we need to reset the cache. This might be a rare case, but the event that we have a false read of a tag, we don't want to permanently store it in the cache, then use that cache always (resulting in it trying to find a non-existent tag).

In any case, I did it just to flush out info, mostly to avoid storing an old cache (from say, an earlier Tag identification)

Reply:
aren't false positives like one in life time rare events? even so 20 frames is fairly quick to clear that. i think it'll clean up a lot of this by just using the cache & only setting it to left target (or right) if it sees smth else increase misses, & if the misses are above the threshold reset it, & then everywhere else only use the cache.